app/src/main/java/com/babylon/wallet/android/presentation/dapp/unauthorized/accountonetime/OneTimeChooseAccountsViewModel.kt

```diff
@@ -85,23 +87,27 @@ class OneTimeChooseAccountsViewModel @Inject constructor(
                 )
             }
         }
-        val isContinueButtonEnabled = if (_state.value.isExactAccountsCount) {
-            _state
-                .value
-                .availableAccountItems
+        _state.update {
+            it.copy(isContinueButtonEnabled = isSelectionCompleted(state.value.availableAccountItems))
+        }
+    }
+
+    private fun isSingleChoice(): Boolean {
+        return args.numberOfAccounts == 1 && args.isExactAccountsCount
+    }
+
+    private fun isSelectionCompleted(availableAccountItems: List<AccountItemUiModel>): Boolean {
+        return if (args.isExactAccountsCount) {
+            availableAccountItems
                 .count { accountItem ->
                     accountItem.isSelected
                 } == args.numberOfAccounts
         } else {
-            _state
-                .value
-                .availableAccountItems
+            availableAccountItems
                 .count { accountItem ->
                     accountItem.isSelected
                 } >= args.numberOfAccounts
         }
-
```

Comment:
In general such checks can be done automatically by a backing field inside state. Since state has immutable elements, we could expose:
```
val isContinueEnabled: Boolean
     get() = // put the code here that checks the selected accounts are fulfilling the request based on args
                 // given that args are included in state (imho should be)
```                 
So you can freely update the state and you don't have to worry on forgetting to also call the `isSelectionCompleted()` method.

Reply:
Good point. Thanks!